File: 프로그래머스/level1_2016년.py
def solution(a, b):
    day = ["THU", "FRI", "SAT", "SUN", "MON", "TUE", "WED"]
    # 1월 1일이 금요일이므로, 7로 나눴을 때 인덱스 1이 금요일로 나오게 배열 설정
    mon = [31, 29, 31, 30, 31, 30, 31, 31, 30, 31, 30, 31]

    # mon배열의 (a-1)월까지 더한 수에
    return day[(sum(mon[:a-1]) + b) % 7]

print(solution(5, 24))

File: 프로그래머스 복습(java)/src/level1_2016년.java
public class level1_2016년 {
    static class Solution {
        public String solution(int a, int b) {
            String[] answer = {"THU", "FRI", "SAT", "SUN", "MON", "TUE", "WED"};
            int[] month = {0, 31, 29, 31, 30, 31, 30, 31, 31, 30, 31, 30, 31};
            int days = 0;
            for (int i = 1; i < a; i++) {
                days += month[i];
            }
            days += b;
            return answer[days % 7];
        }
    }

    public static void main(String[] args) {
        Solution solution = new Solution();
        int a = 5, b = 24;
        System.out.println(solution.solution(a, b));
    }
}
